### ISSUE TITLE
Documentation is lacking and is hard to navigate/search

### ISSUE THREAD

**Issue by backspacerhino** [questdb/questdb#589]
**Is your feature request related to a problem? Please describe.**
I think that the current documentation is seriously lacking some base things ex. how to make a field nullable.
Furthermore, it is really hard to navigate/search the documentation in sense that I often end up to a completely different section from what I wanted.

**Describe the solution you'd like**
I would like to be able to easily search the documentation 

**Describe alternatives you've considered**
Using other time series database

**Additional context**
None

**Comment by bluestreak01** [comment]
All column types are implicitly nullable with exception of `byte`, `char` and `short`. There is no explicit nullability flag.

Could you share examples of what were you looking for and where you ended up with the navigation?

**Comment by backspacerhino** [comment]
Given the a simple example:

```
CREATE TABLE test (
    ts timestamp,
    name string,
    age int
) timestamp (ts)
```

When I connect using [NodeJS example](https://questdb.io/docs/tutorial/nodejs) how should I insert data? I would assume that instead of .query() I would use .insert() but that seems to not be the case.

So when I try to insert using this:

await client.query(`INSERT INTO test (ts,age) VALUES ('${ data.ts }',100);`)

I don't get an error but I don't get an insert but when I use 

await client.query(`INSERT INTO test (ts,age) VALUES (to_timestamp('${ data.ts }', 'yyyy-MM-dd HH:mm:ss'),100);`)

then I do get an insert.

**Note:** `data.ts` is ex. "2020-09-05 11:24:55"

I've also noticed that I am unable to explicitly insert a null value.

Regarding the navigation, I feel like GUIDES, API and TUTORIALS pages have mixed contents and it should be separated more obviously.

So to sum up:

1.  More detailed description on creating tables and querying/inserting data
2. If possible, allow explicit null fields insert
3. More examples on specific usages such as NodeJS

**Comment by jaugsburger** [comment]
@backspacerhino Thanks for the examples

Regarding using `client.insert()` for inserting data in nodejs, I had a look in the official [node-postgres API](https://node-postgres.com/api/client) documentation and there isn't a method `insert()`, all examples use `query()`. We have a link to the node-postgres API in the tutorial page for reference.

For inserting null, you can omit the column, like you did in your example. E.g.: `INSERT INTO test (ts,age) VALUES (to_timestamp('${ data.ts }', 'yyyy-MM-dd HH:mm:ss'),100);`

For explicit insertion of null values, we currently don't support it for all data types but we will be working on that soon. Link to issue: https://github.com/questdb/questdb/issues/592

Also, you had issues inserting this row: `INSERT INTO test (ts,age) VALUES ('2020-09-05 11:24:55',100);`. This is because the string used for the timestamp needs to be converted to type `timestamp`. See doc here: https://questdb.io/docs/reference/function/date-time#to_timestamp.

About your concluding points, I will be adding more content for the nodejs tutorial as well as other languages.

**Comment by mpsq** [comment]
@backspacerhino we've been working on a big overhaul of the documentation. We took your feedback into consideration and simplified its structure drastically (at least we hope!). In particular:

- Thorough examples for the different APIs with popular languages/tools.
- Remove duplicates (basically merging guides and tutorials in `develop`)
- Flatter / "more sensible" navigation
- Tons of other changes and improvements, I will let you go through it!

We are super excited about this, we hope it will make a difference.

I will close this issue for now, I invite you to look at the documentation preview and give feedback if you find anything. For any other friction points, please open a new issue :).

The documentation preview is accessible here:
https://deploy-preview-155--jolly-turing-362f10.netlify.app/docs/introduction/

(note that this is still WIP, wording might change in the coming days)

**Comment by mpsq** [comment]
EDIT: Use this link: https://5f85a67f36fbb600087fbdc6--jolly-turing-362f10.netlify.app/
